packages/protvista-structure/src/protvista-structure.ts

```diff
@@ -0,0 +1,309 @@
+
+      switch (attrName) {
+        case "pdb-id":
+          if (newVal !== null) {
+            this._pdbID = newVal;
+            this.selectMolecule(this._pdbID);
+          }
+          break;
+
+        case "accession":
+          this._accession = newVal;
+          break;
+
+        case "highlight":
+          this._highlight = ProtvistaStructure._parseHighlight(
+            this.getAttribute("highlight")
+          );
+          break;
+
+        case "height":
+          this._height = newVal;
+          break;
+
+        default:
+          break;
+      }
+
+      this._planHighlight();
+    }
+  }
+
+  _planHighlight(): void {
+    // If rendering is already planned, skip the rest
+    if (this._plannedRender) {
+      return;
+    }
+    // Set a flag and _planRender at the next frame
+    this._plannedRender = true;
+    requestAnimationFrame(() => {
+      // Removes the planned rendering flag
+      this._plannedRender = false;
+
+      if (!this._selectedMolecule) {
+        return;
+      }
+
+      this.highlightChain();
+    });
+  }
+
+  async loadPDBEntry(pdbId: string): Promise<unknown> {
+    try {
+      const data = await fetch(
+        `https://www.ebi.ac.uk/pdbe/api/mappings/uniprot/${pdbId}`
+      );
+      return await data.json();
+    } catch (e) {
+      this._structureViewer.showMessage("Error", `Couldn't load PDB entry`);
+      throw new Error(e);
+    }
+  }
+
+  async selectMolecule(id: string): Promise<void> {
+    const pdbEntry = await this.loadPDBEntry(id);
+    const mappings =
+      Object.values(pdbEntry)[0].UniProt[this._accession]?.mappings;
+    await this._structureViewer.loadPdb(id.toLowerCase());
+    this._selectedMolecule = {
+      id,
+      mappings,
+    };
+    this._planHighlight();
+  }
+
+  propagateHighlight(sequencePositions: number[]): void {
+    // sequencePositions assumed to be in PDB coordinate space
+    if (
+      !sequencePositions?.length ||
+      sequencePositions.some((pos) => !Number.isInteger(pos))
+    ) {
+      return;
+    }
+
+    let translated;
+    try {
+      translated = sequencePositions.map((pos) =>
+        translatePositions(pos, pos, this._selectedMolecule.mappings, "PDB_UP")
+      );
+    } catch (error) {
+      if (error instanceof PositionMappingError) {
+        this._structureViewer.showMessage("Error", error.message);
+        return;
+      }
+      throw error;
+    }
+    const highlight = translated
+      .filter(Boolean)
+      .map((residue) => `${residue.start}:${residue.end}`);
+    this.setAttribute("highlight", highlight.join(","));
+    const event = new CustomEvent("change", {
+      detail: {
+        highlight,
+      },
+      bubbles: true,
+      cancelable: true,
+    });
+    this.dispatchEvent(event);
+  }
+
+  highlightChain(): void {
+    this._structureViewer.clearMessages();
+    if (!this._highlight) {
+      this._structureViewer.clearMessages();
```

Comment:
Don't think you need to clear the messages twice?

Reply:
Debugging relic - removed